File: documentation/Exercise 3/Java/src/main/java/de/thu/recipebook/activities/RecipeListActivity.java
package de.thu.recipebook.activities;

import android.os.Bundle;
import android.widget.ListView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;
import java.util.List;

import de.thu.recipebook.R;
import de.thu.recipebook.models.Recipe;
import de.thu.recipebook.models.RecipeCollection;
import de.thu.recipebook.models.RecipeListAdapter;

public class RecipeListActivity extends AppCompatActivity {
    private RecipeCollection recipeCollection;
    private RecipeListAdapter adapter;

    private ListView listView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_recipe_list);

        recipeCollection = RecipeCollection.getInstance();
        adapter = new RecipeListAdapter();
        List<Recipe> data = recipeCollection.getRecipes();
        adapter.setData(data);

        listView = findViewById(R.id.list_view_recipe);
        listView.setAdapter(adapter);
    }
}

File: documentation/Exercise 3/Kotlin/src/main/java/de/thu/recipebook/activities/RecipeListActivity.kt
package de.thu.recipebook.activities

import android.os.Bundle
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import de.thu.recipebook.R
import de.thu.recipebook.models.Recipe
import de.thu.recipebook.models.RecipeCollection
import de.thu.recipebook.models.RecipeListAdapter

class RecipeListActivity : AppCompatActivity() {
    private var recipeCollection: RecipeCollection? = null
    private var adapter: RecipeListAdapter? = null

    private var listView: ListView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_recipe_list)

        recipeCollection = RecipeCollection.instance
        val recipeNames: MutableList<String> = ArrayList()
        for (recipe in recipeCollection?.getRecipes()!!) {
            recipeNames.add(recipe.name)

        }

        adapter = RecipeListAdapter()
        val data: List<Recipe> = recipeCollection!!.getRecipes()
        adapter!!.setData(data)

        listView = findViewById(R.id.list_view_recipe)
        listView?.adapter = adapter
    }
}